components/style/custom_properties.rs

```diff
@@ -106,10 +107,16 @@ where
 
     /// Insert a new key-value pair.
     pub fn insert(&mut self, key: K, value: V) {
-        if !self.values.contains_key(&key) {
-            self.index.push(key.clone());
+        let OrderedMap { ref mut index, ref mut values } = *self;
+        match values.entry(key) {
+            Entry::Vacant(entry) => {
+                index.push(entry.key().clone());
+                entry.insert(value);
+            }
+            Entry::Occupied(mut entry) => {
```

Comment:
Interesting! This [doesn't need `mut`](https://doc.rust-lang.org/std/collections/hash_map/struct.OccupiedEntry.html#method.insert)?

Reply:
Yeah, I had another commit on top (which I want to submit separately since it's non-trivial) that fixed this, whoops.